staging/src/k8s.io/apimachinery/pkg/util/httpstream/wsstream/conn.go

```diff
@@ -178,7 +179,11 @@ func (conn *Conn) Open(w http.ResponseWriter, req *http.Request) (string, []io.R
 	go func() {
 		defer runtime.HandleCrash()
 		defer conn.Close()
-		websocket.Server{Handshake: conn.handshake, Handler: conn.handle}.ServeHTTP(w, req)
+		handler := conn.handle
+		if _, ok := conn.protocols[remotecommand.StreamProtocolV5Name]; ok {
+			handler = conn.handleV5
+		}
+		websocket.Server{Handshake: conn.handshake, Handler: handler}.ServeHTTP(w, req)
```

Comment:
hmm, but it seems you are right Arda, we should have proxy support for the clients , and the golang library does not have it

Reply:
gorilla solves this

Reply:
The PR description has a `NOTE` at the bottom, which details why Gorilla websockets library was chose for the websockets client, since everyone will want to know why the library was chosen. I added one more reason with the direct support for proxies on the client.